vulture_os/toolkit/api_parser/hornetsecurity/hornetsecurity.py

```diff
@@ -0,0 +1,361 @@
+        """
+        Get root hierarchy id to use it as an 'object_id' later to gather logs
+        """
+        resp = self.session.get(
+            url = self.HIERARCHY_GET,
+            params = {"limit": 1, "offset": 0},
+            headers = {"Authorization": f"Token {self.hornetsecurity_token}"})
+        try:
+            json_resp = resp.json()
+        except Exception as e:
+            raise HornetSecurityAPIError(f"Failed to decode json response - {resp.content} -- {e}")
+
+        hierarchy = json_resp.get('hierarchy', {})
+        if hierarchy[0] if hierarchy else None:
+            if hierarchy[0].get('id'):
+                return hierarchy[0]['id']
+
+        msg = f"Fail to get hierarchy id -- {resp.content}"
+        logger.exception(f"[{__parser__}][get_root_hierarchy]: {msg}", extra={'frontend': str(self.frontend)})
+        raise HornetSecurityAPIError(msg)
+
+
+    def execute_query(self, kind: str, since: datetime, to: datetime, limit: int, offset: int) -> tuple[str, int, list]:
+        since = datetime.strftime(since, "%Y-%m-%dT%H:%M:%SZ")
+        to = datetime.strftime(to, "%Y-%m-%dT%H:%M:%SZ")
+
+        retry = 3
+        while retry > 0:
+            try:
+                if kind == "auditing":
+                    resp = self.session.post(
+                        url = self.AUDITS_SEARCH,
+                        params = {"object_id": self.root_id},
+                        headers = {"APP-ID": self.hornetsecurity_app_id, "Authorization": f"Token {self.hornetsecurity_token}"},
+                        data = {"date_from": since, "date_to": to, "limit": limit, "offset": offset},
+                        timeout = 10)
+                elif kind == "emails":
+                    resp = self.session.post(
+                        url = self.EMAILS_SEARCH,
+                        params = {"object_id": self.root_id},
+                        headers = {"APP-ID": self.hornetsecurity_app_id, "Authorization": f"Token {self.hornetsecurity_token}"},
+                        data = {"date_from": since, "date_to": to, "limit": limit, "offset": offset},
+                        timeout = 10)
+
+            except Exception as e:
+                logger.exception(f"[{__parser__}][execute_query]: Unexpected error while fetching {kind}'s logs - {since, to} -- {e}", extra={'frontend': str(self.frontend)})
+                retry -= 1
+                continue
+
+            try:
+                json_resp = resp.json()
+            except Exception as e:
+                logger.exception(f"[{__parser__}][execute_query]: Failed to decode json response {resp.content} -- {e}", extra={'frontend': str(self.frontend)})
+                raise HornetSecurityAPIError(e)
+
+            # Handles more than 10k logs to retrieve in a single query
+            toRetrieve = json_resp.get('num_found_items', 0)
+            if toRetrieve >= 10000: # error_id returned by API in this case is 3089
+                return "divide", toRetrieve, []
```

Comment:
the value of `num_found_items`/`toRetrieve` is enough in code to trigger a new execution with less time range, why are you adding this third parameter? (and returning a string for internal conditions is less-than-optimal)